### PR TITLE
Admin interface typescript

### PR DESCRIPTION
**Pull request by jlyon1**
An overhaul of the admin interface in typescript, implements the following for Routes, Stops, and Vehicles:

-Create
-Edit
-Delete
-View

*Note:* Stops not complete yet, I plan on finishing tomorrow.

The goal here is to simplify the structure of the admin interface, and have better error displays for the user, so if an action fails, the user will be appropriately notified. 

Usability as well as functionality and implementation should be reviewed.

### CODE REVIEW

File: frontend/src/components/admin/routeEditing.vue [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        setPoints(points: Array<{latitude: number, longitude: number}>) {
+            this.route.points = points;
+        },
+        send() {
+            this.sending = true;
+            if (!this.formValid) {
+                return;
+            }
+            if (!this.creation) {
+                AdminServiceProvider.EditRoute(this.route).then(() => {
```
Comment (garoller) [inline frontend/src/components/admin/routeEditing.vue]: Editing/creating routes seems buggy, and I will follow-up with more details.

File: frontend/src/Admin.vue [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<template>
+  <div id="admin">
+    <nav class="navbar" role="navigation" aria-label="main navigation">
+      <div class="navbar-brand">
+        <a class="navbar-item" href="https://shuttles.rpi.edu">
+          <p>Shuttle Tracc - <span class="has-text-danger">Fleet Management</span></p>
```
[the remark this answers is not in the document]
Comment (jlyon1) [reply frontend/src/Admin.vue]: no

File: frontend/src/components/admin/stopsAdmin.vue [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<template>
```
[the remark this answers is not in the document]
Comment (jlyon1) [reply frontend/src/components/admin/stopsAdmin.vue]: yeah, stop functionality is not in this PR, i'll end up doing it in another PR I think unless it's a major issue, most of what P&T wants to do is add messages.

File: frontend/src/components/admin/routeEditing.vue (hunk 1 above)
Comment (kochman) [reply frontend/src/components/admin/routeEditing.vue]: The schedule constraint requires that if `start_day = end_day` then `start_time < end_time`.

Here's [the actual constraint](https://github.com/wtg/shuttletracker/blob/bfa68d4b481b909d7b30c2edfcfdf79f23510076/postgres/route.go#L52):
```
-- Note: active intervals for route schedules for a route cannot wrap around
-- the week boundary. This is for simplicity of implementation in the
-- route_is_active() function.
CHECK (
	(start_day = end_day AND start_time < end_time) OR (start_day < end_day)
)```

**Comment by jlyon1** [comment]
I chose to disable the map on the route editing interface as you cannot edit the route through that interface anyway. Should make things less confusing, eventually we can change the flow of route editing to make more sense

**Pull request merged**